substrate/substrate-dump/src/dumper.ts

```diff
@@ -193,11 +201,14 @@ export class Dumper {
                 }
             }
         } else {
-            let archive = new ArchiveLayout(this.fs())
+            const archive = new ArchiveLayout(this.fs())
+            const prometheus = this.prometheus();
+            if (this.options.metricsPort) await prometheus.serve();
             await archive.appendRawBlocks({
                 blocks: (nextBlock, prevHash) => this.saveMetadata(this.process(nextBlock, prevHash)),
                 range: this.range(),
-                chunkSize: this.options.chunkSize * 1024 * 1024
+                chunkSize: this.options.chunkSize * 1024 * 1024,
+                onSuccessWrite: this.options.metricsPort ? ((block) => { prometheus.setLastSavedBlock(block); }) : undefined
```

Comment:
```ts
{
  onSuccessWrite: block => this.prometheus().setLastSavedBlock(block)
}
```

No need for availability check.

Reply:
Well, agree, but there is also no need to set metric if it is not used